src/search/dtos/search-all-help-me-boards.dto.ts

```diff
@@ -0,0 +1,81 @@
+import { HelpMeBoard } from 'src/boards/entities/help-me-board.entity';
+import { CategoryList } from 'src/common/entity/category-list.entity';
+import { SearchUserDto } from './search-user-dto';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class SearchAllHelpMeBoardsDto
+  implements
+    Omit<
+      HelpMeBoard,
+      | 'categoryList'
+      | 'categoryId'
+      | 'updatedAt'
+      | 'user'
+      | 'userId'
+      | 'helpMeBoardImages'
+    >
+{
+  @ApiProperty({
+    description: '도와주세요 게시글 고유 id',
+    format: 'integer',
+  })
+  id: number;
+
+  @ApiProperty({
+    description: '도와주세요 게시글 제목',
+  })
+  head: string;
+
+  @ApiProperty({
+    description: '도와주세요 게시글 본문. 30글자까지 자름',
+    maxLength: 30,
+  })
+  body: string;
+
+  @ApiProperty({
+    description: '작성자 고유 id',
+  })
+  createdAt: Date;
+
+  @ApiProperty({
+    description: '게시글 작성자 정보를 매핑한 객체',
+  })
+  user: SearchUserDto;
+
+  @ApiProperty({
+    description: '도와주세요 게시판의 이미지 배열',
+    type: 'array',
+    items: {
+      type: 'object',
+      properties: {
+        imageUrl: {
+          type: 'string',
+          description: '도와주세요 게시판의 이미지 url. 한개만 가져옴',
+        },
+      },
+    },
+  })
+  helpMeBoardImages: { imageUrl: string }[] | [];
```

Comment:
```suggestion
  helpMeBoardImages: [{ imageUrl: string }];
```
한개만 가져온다면 다음처럼 타이핑을 할 수 있습니다,

Reply:
한개만 가져오거나 빈 배열을 가져오는데 이렇게 했을 시에 타입 에러가 나네요 ㅠㅠ


'[HelpMeBoard[], User[]]' 형식은 '[SearchAllHelpMeBoardDto[], SearchAllMentorDto[]]' 형식에 할당할 수 없습니다.
  소스의 0 위치에 있는 형식이 대상의 0 위치에 있는 형식과 호환되지 않습니다.
    'HelpMeBoard[]' 형식은 'SearchAllHelpMeBoardDto[]' 형식에 할당할 수 없습니다.
      'HelpMeBoard' 형식은 'SearchAllHelpMeBoardDto' 형식에 할당할 수 없습니다.
        'helpMeBoardImages' 속성의 형식이 호환되지 않습니다.
          'HelpMeBoardImage[]' 형식은 '[{ imageUrl: string; }] | []' 형식에 할당할 수 없습니다.
            'HelpMeBoardImage[]' 형식은 '[]' 형식에 할당할 수 없습니다.
              대상에서 0개 요소만 허용하지만, 소스에 더 많이 있을 수 있습니다.ts(2322)